src/select.cr

```diff
@@ -0,0 +1,112 @@
+module Select
+  # A select target will produce a select action when select is called.
+  #
+  # :nodoc:
+  module Action
+    abstract def execute
+
+    abstract def activate
+    abstract def deactivate
+    abstract def owns_token?(resume_token : Void*)
+
+    # Subcategory of select actions which can be checked for readiness using a
+    # method call.
+    module Checkable
+      include Action
+
+      abstract def ready?
+    end
+
+    # Subcategory of select actions which can be polled using the `poll`
+    # syscall.
+    module Pollable
+      include Action
+
+      abstract def pollfd : LibC::Pollfd
+    end
+  end
+
+  def self.select(*actions : Action)
+    self.select(actions)
+  end
+
+  def self.select(actions : Tuple | Array, has_else = false)
+    pollfds = Array(LibC::Pollfd).new
+    fdmap = Hash(Int32, Int32).new # Map of poll file descriptor to action index
```

Comment:
Maybe one could recycle the `pollfds` and `fdmap` variables? Chances are good, that a Fiber is always running the same `select` in a loop, with the same arguments, thus the sizes of both would never change and no sub-sequent allocations will be done. That, or we need to introduce VLAs :)